plugins/prco/src/dataSources.ts

```diff
@@ -0,0 +1,508 @@
+import type { ManagedCollectionFieldInput } from "framer-plugin"
+import * as v from "valibot"
+import {
+    AlexaSchema,
+    ClippingImageAsImageSchema,
+    ClippingsSchema,
+    FeaturedImagesForClSchema,
+    ImageSizesSchema,
+    MediaInfoSchema,
+    MediaKitsSchema,
+    type PrCoItem,
+    PressReleasesSchema,
+    SocialSchema,
+    TagsSchema,
+} from "./api-types"
+
+const API_URL = "https://api.pr.co/v1"
+
+export interface PrCoDataSource {
+    id: string
+    name: string
+    /**
+     * The fields of the data source.
+     *
+     * The first field is the ID field.
+     * The rest of the fields are the fields of the data source.
+     */
+    fields: readonly PrCoField[]
+    fetch: (pressRoomId: string) => Promise<PrCoItem[]>
+}
+
+async function fetchPrCoData(url: string): Promise<unknown> {
+    try {
+        const response = await fetch(url)
+        return await response.json()
+    } catch (error) {
+        console.error("Error fetching PrCo data:", error)
+        throw error
+    }
+}
+
+export type PrCoField = ManagedCollectionFieldInput & {
```

Comment:
Must this be an intersection? If the `ManagedCollectionFieldInput` structure were just a standalone property in `PrcoField` then `removePrCoKeys` wouldn't be needed at all. You can make exceptions for collections with `Exclude<ManagedCollectionFieldInput, { type: ... }>` and such.

Also, I think you should try removing `removePrCoKeys` regardless, since excess properties passed to the API should be okay.